### ISSUE TITLE
Crash when a thread has been opened in a new window

### ISSUE THREAD

**Issue by siddhpant** [flathub/com.slack.Slack#280]
Basically the title.

It happens randomly, and I don't know how to reproduce it.

**Comment by StefanWerW** [comment]
Same here

**Comment by ZVNexus** [comment]
Was running into the same thing on the snap, but can't reproduce anymore after 4.38.121. Reopen if it's still an issue.

**Comment by siddhpant** [comment]
(I don't know if this is a new bug or not, but it is definitely related)

I seem to hit the crash even now.

In fact, I found another way for crash - when a thread has been opened and there is a draft,  maximizing the slack window. It can miss but it hits many times. Maybe it is a race condition in slack code? I suspect a similar thing was responsible for the original new window crashes, as it could be related to something with the window during loading or rendering:

```
$ flatpak run com.slack.Slack
Initializing local storage instance
(electron) Sending uncompressed crash reports is deprecated and will be removed in a future version of Electron. Set { compress: true } to opt-in to the new behavior. Crash reports will be uploaded gzipped, which most crash reporting servers support.
Gtk-Message: 11:29:28.429: Failed to load module "canberra-gtk-module"
Gtk-Message: 11:29:28.434: Failed to load module "canberra-gtk-module"
LaunchProcess: failed to execvp:
xdg-settings
[51:0520/112941.543278:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
/app/bin/com.slack.Slack: line 12:     4 Segmentation fault      env TMPDIR=${XDG_CACHE_HOME} XDG_DOWNLOAD_DIR=$(xdg-user-dir DOWNLOAD) zypak-wrapper /app/extra/slack -s ${EXTRA_ARGS} "$@"
```

Basically, I am able to reproduce crash after a crash:
1. In a thread, type something but do not send. (This might not be needed but it helps)
2. Optional - Open thread in a new window.
3. Maximise the window (if a new window or new start, do it when loading).
4. If crash, then you can reproduce it again.
5. If you do wait for some time or do something, send or remove msg, or close thread it won't happen.

**Comment by siddhpant** [comment]
Crash only happens with the X11 socket. When switched to Wayland socket, I can no longer reproduce it (I'm on Wayland).

The only issue is I see light theme header bar on Wayland socket, which was why I was using the X11 socket.

**Comment by siddhpant** [comment]
Now I seem to get "not responding" after some time.

![image](https://github.com/flathub/com.slack.Slack/assets/25429745/818f4acb-ec06-4ba4-9d03-b6846c3df33c)

**Comment by siddhpant** [comment]
If I install deb from their website directly, I get a segmentation fault by just maximising.

**Comment by colorfulmoth** [comment]
> Crash only happens with the X11 socket. When switched to Wayland socket, I can no longer reproduce it (I'm on Wayland).
> 
> The only issue is I see light theme header bar on Wayland socket by default, which was why I was using the X11 socket. Edit: I have fixed that now by exporting the theme variable.

This fixed the crash for me too. For anyone too lazy to look up the "fix", this is the command I used to switch to the Wayland socket.

`flatpak override --socket=wayland com.slack.Slack`

**Comment by ZVNexus** [comment]
> This fixed the crash for me too. For anyone too lazy to look up the "fix", this is the command I used to switch to the Wayland socket.
> 
> `flatpak override --socket=wayland com.slack.Slack`

Just keep in mind that doing that introduces other regressions: https://github.com/flathub/com.slack.Slack/issues/183#issuecomment-1902686110

**Comment by ZVNexus** [comment]
Also keeping this issue closed since it seems to be a result of upstream and not the Flatpak itself. Please report these issues to Slack directly.

**Comment by siddhpant** [comment]
Yep, I did. They said it will be fixed in the next version.
